### DISCUSSION TITLE
How to Upload a File to Database using Prisma

### DISCUSSION THREAD

**Discussion by codeDev911** [prisma/prisma#11795] [Q&A] [answered]
Hello Developers 
I'm new to Prisma . I want to save files in DataBase Using Prisma 
How i can do this ??

`model user{
  id Int @id @default(autoincrement())
  name String
  email String @unique
  // What i do for image file or other files
  profilePic ??? // What i Write here to save Imgae as well

}`

**Comment by borngraced** [comment, marked as answer]
You'll need a cloud storage e.g "google cloud staorage" to store the uploaded files then  you'll update the profilePic field with the uploaded file link.
So you're mostly going to be using something like this instead

```model user{
id Int @id @default(autoincrement())
name String
email String @unique
profilePic String?
}

**Comment by bluefield-creator** [comment]
Cloud storage or custom storage. There are different approaches. You could go hardcore and save the file as a base64 string of it, and then render the image on the client.

Best approach: Cloud Storage
Alternatives: Store in the server
Fastest Approach: Base64 Strings

To note; **Prisma is not a database**, it's an **ORM**, meaning it handles the **connection** between **you and your database**, I've seen some databases support file uploads, but I doubt prisma supports them. (As it primarily works only with Mongo & SQL databases)

So quick lesson of good practices: When a user uploads an image, you reference the image name, not the image file itself.

```
user -- Uploads image --> Handle Image Upload -- Save Image with custom name --> Save image name to database
````

```
user -- Loads page --> Require image -- search in the database for the image name --> Return image name so the client can load it in the browser / client application.
```

**Reply by davidjgonzalez** [reply]
Do you have thoughts on a flow to store protected content in cloud storage? Meaning a user uploads a image, and only that user (or other users that owner has shared the images worth) can see it.

Do we assume the cloud storage URL to unguessable enough (security through obscurity)? What if someone somehow grabs the url?

It seems it could be nightmarish trying to maintain file permissions on a cloud storage file?

**Reply by bluefield-creator** [reply]
Most of the times, cloud storage providers will offer some sort of permission management service. I am not entirely sure how it works but you could probably find the answer within the documentation.

**Reply by bluefield-creator** [reply]
Though, keep in mind hash filenames are often more than enough.

**Reply by bluefield-creator** [reply]
Highly recommend you check [this stackoverflow thread.](https://stackoverflow.com/questions/9296272/restricting-files-from-google-cloud-storage-to-the-users-that-have-authenticated)

**Comment by aaronamendez** [comment]
Great walkthrough!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
